lib/src/modules/gates.dart

```diff
@@ -724,3 +724,57 @@ class IndexGate extends Module with InlineSystemVerilog {
     return '$target[$idx]';
   }
 }
+
+/// A Replication Operator [Module].
+///
+/// It takes two inputs (bit and width) and outputs a [Logic] representing
+/// the input bit repeated over the input width
+class ReplicationOp extends Module with InlineSystemVerilog {
+  late final String _inputName;
+  late final String _outputName;
+  // Width of the output signal
+  late final int _width;
+
+  /// The primary input to this gate.
+  Logic get _input => input(_inputName);
+
+  /// The output of this gate.
+  Logic get replicated => output(_outputName);
+
+  /// Constructs a ReplicationOp
+  ///
+  /// The bit [bit] will be repeated over the [_width] as an output.
+  /// [Module] is in-lined as SystemVerilog, it will use {width{bit}}
+  ReplicationOp(Logic bit, this._width) : super() {
```

Comment:
You don't need `super()`, it's implicit

You don't need to make `_width` `late`, since it's set in the constructor.

I know other code in here follows this same style, but you don't need to make `_inputName` and `_outputName` `late` either, you could set them as part of the constructor:
```dart
ReplicationOp(Logic bit, this._width) : _inputName = '...', _outputName = '...';
```